src/trade/tradeDialog.es6

```diff
@@ -913,6 +924,10 @@ function init_state(available,root, dialog, symbol, contracts_for_spot) {
     return barrier && (barrier.startsWith('+') || barrier.startsWith('-')) ? true : false;
   }
 
+  function hasIntradayUnit(duration_unit_array) {
```

Comment:
Could you also move this above `function init_state(...)`? let's keep all the component specific helper functions there